plans/004-isolate-web-bridge-content-world.md

```
@@ -0,0 +1,232 @@
+
+1. **Feature parity**: open a real webpage in web reading mode; confirm text selection → highlight popover → highlight creation and click-to-edit still work, and the AI panel still receives page text (send any message and confirm no "no page text" degradation). Reopen the page to confirm the highlight re-anchors.
+2. **Isolation**: create `/tmp/vellum-bridge-test.html` containing a script that does
+   `try { window.webkit.messageHandlers.vellum.postMessage({vellum:true,type:"navigate",url:"https://example.com"}); document.title="BRIDGE-REACHABLE" } catch(e) { document.title="BRIDGE-ISOLATED" }`
+   Serve it (`python3 -m http.server` in /tmp) and open `http://localhost:8000/vellum-bridge-test.html` in web reading mode. Expected: the tab does **not** navigate away, and the page title reads `BRIDGE-ISOLATED`.
+3. **Offline path**: open a page, let the auto-archive complete, kill network access to it (or reopen while offline), and confirm the snapshot renders **with** working highlights (the user script injects on snapshot navigations too).
+
+If you cannot drive the GUI in your environment, report exactly which of these three checks you could not run — do not claim them.
+
+## Test plan
+
+The isolation property is inherently behavioral (it lives in WebKit), so the primary tests are step 5's checks. Additionally: if `prepareHtml` is unit-testable as a pure function (it is — static, string in/out), add `Tests/WebHtmlPrepareTests.swift` asserting the output of `prepareHtml("<html><head></head></html>", pageUrl:..., offline: false) `contains `<base href` and does **not** contain `<script`. Model on `Tests/ScratchpadImportTests.swift`.
```

markdownlint-cli2 (0.23.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)